src/Model/Webhook/Response.php

```diff
@@ -126,14 +127,28 @@ public function createContextFromSession($name, array $data = []) {
     }
 
     public function jsonSerialize() {
-        // Prepend fulfillmentText.
-        $json = ['fulfillmentText' => $this->getFulfillment()->getText()];
-
-        // Append fulfillmentMessages, if any.
+        $json = [];
+        if ($text = $this->getFulfillment()->getText()) {
+          $json['fulfillmentText'] = $this->getFulfillment()->getText();
+        }
         if ($messages = $this->getFulfillment()->getMessages()) {
-            $json['fulfillmentMessages'] = array_map(function($message) {
-                return $message->jsonSerialize();
-            }, $messages);
+          $json['fulfillmentMessages'] = [
```

Comment:
This shouldn't be hardcoded. We should have a Message class implementing `JsonSerializable()` so the logic in this block is delegated to Messages. Same for Event.